scripts/generate_html_report.py

```diff
@@ -0,0 +1,239 @@
+"""
+This script processes SVG files in a specified folder to generate an HTML report.
+The report compares the current SVG icons with their previous versions and displays their differences.
+
+Usage:
+1. Ensure that the SVG files are organized in the specified folder, with a 'previous' subfolder containing the previous versions of the SVGs.
+2. Run the script: `python script_name.py`
+3. When prompted, enter the path to the folder containing the SVG files.
+4. The script will parse the SVG files and generate an HTML report saved as 'icons_data.html' in the specified folder.
+
+Requirements:
+- The script assumes that the SVG files follow the naming pattern: 'ic_fluent_<name>_<size>_<style>.svg'
+- Ensure the 'previous' subfolder is present in the specified folder and contains the previous versions of the SVG files.
+
+Example SVG Folder Structure:
+  /path/to/svgs/
+  ├── ic_fluent_icon1_16_regular.svg
+  ├── ic_fluent_icon1_16_filled.svg
+  ├── ic_fluent_icon2_20_regular.svg
+  ├── previous/
+  │   ├── ic_fluent_icon1_16_regular.svg
+  │   ├── ic_fluent_icon1_16_filled.svg
+  │   └── ic_fluent_icon2_20_regular.svg
+
+The generated HTML report will display:
+- The name of the icons
+- The available sizes
+- The regular and filled styles of the icons
+- A visual comparison between the current and previous versions of the icons
+
+Note:
+- The HTML report includes buttons to toggle the visibility of new icons and to zoom in/out on the icons.
+"""
+
+import os
+import re
+
+# Prompt the user for the path to the SVGs folder
+svg_folder = input("Please enter the path to the SVGs folder: ")
```

Comment:
can there be a default value to this so one can just hit enter and not have to add a path?